runtime/common/metering.go

```diff
@@ -134,17 +134,20 @@ func NewArrayMemoryUsages(length int) (MemoryUsage, MemoryUsage) {
 		}
 }
 
-func NewArrayAdditionalLengthUsage(originalLength, additionalLength int) MemoryUsage {
-	var newAmount uint64
-	if originalLength <= 1 {
-		newAmount = uint64(originalLength + additionalLength)
+func atreeAdditionalMemoryUsage(originalSize int) uint64 {
```

Comment:
Isn't the calculation based on the new size?

Reply:
Unsure. @fxamacker when splitting, is the worst case based on the depth of the original tree or the depth of the resulting tree?

Reply:
> when splitting, is the worst case based on the depth of the original tree or the depth of the resulting tree?

The worst case is based on the depth of the original tree.

Reply:
In that case this should be correct.

Reply:
Thanks for clarifying @fxamacker !